### ISSUE TITLE
SC2086 fails to warn after "if" branch

### ISSUE THREAD

**Issue by nkovacs** [koalaman/shellcheck#825]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2086
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

if true; then
    foo="some thing*"
else
    foo="something"
fi

ls $foo
```

#### Here's what shellcheck currently says:

No issues detected!

#### Here's what I wanted or expected to see:

```sh
Line 9:
ls $foo
   ^-- SC2086: Double quote to prevent globbing and word splitting.
```

**Comment by koalaman** [comment]
Finally fixed in #2541
